### PR TITLE
Select a11y enhancements

### PR DESCRIPTION
**Pull request by sarkaaa**
# This Pull Request meets the following criteria:

### For new components:

- [ ] Unit and visual regression tests have been added/adjusted for my new feature
- [ ] New Components are registered in index.js of my project
- [ ] New Components have `d.ts` files and are exported in `index.d.ts`

Closes https://kiwicom.atlassian.net/browse/FEPLT-2176 

<details open>
<summary>:sparkles: <i><h3>Description by Callstackai</h3></i></summary>

This PR introduces accessibility enhancements to various components by adding `ariaLabel` props and updating related documentation.

<details>
<summary><b>Diagrams of code changes</b></summary>

```mermaid
sequenceDiagram
    participant User
    participant InputField
    participant Select
    participant InputGroup
    participant Screen Reader

    Note over InputField,Select: Added aria-label support
    
    User->>InputGroup: Interacts with form components
    InputGroup->>InputField: Passes label as ariaLabel
    InputField-->>Screen Reader: Provides accessibility context
    
    InputGroup->>Select: Passes label as ariaLabel
    Select-->>Screen Reader: Provides accessibility context
    
    Note over InputField,Screen Reader: If no visible label present
    InputField->>Screen Reader: Uses ariaLabel for context
    Select->>Screen Reader: Uses ariaLabel for context
    
    Note over InputGroup: Enhanced error/help tooltip handling
    InputGroup->>InputField: Manages tooltip visibility
    InputGroup->>Select: Manages tooltip visibility
```

</details>

<details>
<summary><b>Files Changed</b></summary>
<table>
<tr><th>File</th><th>Summary</th></tr>
<tr><td><a href=https://github.com/kiwicom/orbit/pull/4543/files#diff-b3afff2226b3d319d27c915ddb49ea0646ebb56bddd6f4da905f6eeb00040537>packages/orbit-components/src/ErrorFormTooltip/Tooltip/types.d.ts</a></td><td>Removed <code>inputSize</code> prop from <code>Props</code> interface.</td></tr>
<tr><td><a href=https://github.com/kiwicom/orbit/pull/4543/files#diff-8148e01eaf2782b50a1af03cc9645607c4a48c6392bbdf7937a5ea0e233c69de>packages/orbit-components/src/ErrorFormTooltip/types.d.ts</a></td><td>Removed <code>inputSize</code> prop from <code>Props</code> interface.</td></tr>
<tr><td><a href=https://github.com/kiwicom/orbit/pull/4543/files#diff-f2e9ecbe159fd887322916e9effec729d5358b645acabee377812acd394dc5da>packages/orbit-components/src/InputField/InputField.stories.tsx</a></td><td>Added <code>ariaLabel</code> prop to the <code>InputField</code> component in the Playground story.</td></tr>
<tr><td><a href=https://github.com/kiwicom/orbit/pull/4543/files#diff-b12bab43f299d4badaa7cc5aec122e2ea2ff4bd7302a9fa076c76b9ff89d0b06>packages/orbit-components/src/InputField/README.md</a></td><td>Updated documentation to include <code>ariaLabel</code> prop.</td></tr>
<tr><td><a href=https://github.com/kiwicom/orbit/pull/4543/files#diff-a22047902ec5dad314f04706476107ebd04df90a6577b33c7fdf40c9887e6355>packages/orbit-components/src/InputField/index.tsx</a></td><td>Added <code>ariaLabel</code> prop to the <code>InputField</code> component.</td></tr>
<tr><td><a href=https://github.com/kiwicom/orbit/pull/4543/files#diff-3e385564e61dd8d7efdbcee542dd8b992a2c37ab1b2dfab626a6e522cfa6567a>packages/orbit-components/src/InputField/types.d.ts</a></td><td>Added <code>ariaLabel</code> prop to <code>Props</code> interface.</td></tr>
<tr><td><a href=https://github.com/kiwicom/orbit/pull/4543/files#diff-c1840fc07cac7fa16b429032b546101ad230db211306df98b0b6b0b965472bcd>packages/orbit-components/src/InputFile/index.tsx</a></td><td>Removed <code>inputSize</code> prop from <code>InputFile</code> component.</td></tr>
<tr><td><a href=https://github.com/kiwicom/orbit/pull/4543/files#diff-9039401357cf53e1a6db8986d0a596abc9694f7b6583bd715a1fc0dd12516f97>packages/orbit-components/src/InputGroup/InputGroup.stories.tsx</a></td><td>Updated <code>DateOfBirth</code> story to include labels for better accessibility.</td></tr>
<tr><td><a href=https://github.com/kiwicom/orbit/pull/4543/files#diff-57fefd5d0ae43e5d10e62df7aa73be8d4237c139be050f358fc3c0a9891f032b>packages/orbit-components/src/InputGroup/index.tsx</a></td><td>Removed <code>size</code> prop and updated how children are rendered.</td></tr>
<tr><td><a href=https://github.com/kiwicom/orbit/pull/4543/files#diff-f49cbd515ec07f8635d1a49ca6e76ea3bb55e892c2802ec77590765b4d79a3af>packages/orbit-components/src/InputGroup/types.d.ts</a></td><td>Removed <code>size</code> prop from <code>Props</code> interface.</td></tr>
<tr><td><a href=https://github.com/kiwicom/orbit/pull/4543/files#diff-20126b7c9184dcfd6338647a40e574b48fc234a828dbd6397344997bf60d1917>packages/orbit-components/src/Select/README.md</a></td><td>Updated documentation to include <code>ariaLabel</code> prop.</td></tr>
<tr><td><a href=https://github.com/kiwicom/orbit/pull/4543/files#diff-61824b9b2a0a9e5ee91f1145d6f2fda89d4bf4351b49ec38337904222f02eb91>packages/orbit-components/src/Select/index.tsx</a></td><td>Added <code>ariaLabel</code> prop to the <code>Select</code> component.</td></tr>
<tr><td><a href=https://github.com/kiwicom/orbit/pull/4543/files#diff-b56db16ebe93360040cfef9b74d42626645cee113d9ea698befa428226e7506c>packages/orbit-components/src/Select/types.d.ts</a></td><td>Added <code>ariaLabel</code> prop to <code>Props</code> interface.</td></tr>

</table>
</details>

*This PR includes files in programming languages that we currently do not support. We have not reviewed files with the extensions `.md`. <a href=https://docs.callstack.ai/introduction>See list of supported languages.</a>*

</details>

### CODE REVIEW

File: packages/orbit-components/src/InputFile/index.tsx [hunk 1]
```diff
@@ -157,7 +157,6 @@ const InputFile = React.forwardRef<HTMLDivElement, Props>((props, ref) => {
           help={help}
           error={error}
           helpClosable={helpClosable}
-          inputSize="normal"
```
Comment (DSil) [inline packages/orbit-components/src/InputFile/index.tsx:160]: This is not a breaking change. If anything, this change can be included in the same commit as the BC for ErrorFormTooltip. But removing const props used internally is not a BC

File: packages/orbit-components/src/ErrorFormTooltip/Tooltip/types.d.ts [hunk 2]
```diff
@@ -9,7 +9,6 @@ export interface Props extends Common.Globals {
   readonly helpClosable: boolean;
   readonly isHelp?: boolean;
   readonly referenceElement?: { current: HTMLElement | null };
-  readonly inputSize?: Common.InputSize;
```
Comment (DSil) [inline packages/orbit-components/src/ErrorFormTooltip/Tooltip/types.d.ts:12]: Commit message mentions `size type` prop but it's actually called `inputSize prop`. The BC message should also mention "prop" instead of "type".

File: packages/orbit-components/src/InputGroup/InputGroup.stories.tsx [hunk 3]
```diff
@@ -70,20 +70,30 @@ const selectOptionsMonths = [
...
-        onBlur={action("onBlurIn")}
-        onFocus={action("onFocusIn")}
-        onChange={action("onChangeIn")}
-      />
-      <Select options={selectOptionsMonths} value={selectValue} placeholder="Month" />
-      <InputField placeholder="YYYY" />
-    </InputGroup>
-  ),
+  render: ({ error, help, selectValue, ...args }) => {
+    const [value, setValue] = React.useState(selectValue);
```
[the remark this answers is not in the document]
Comment (DSil) [reply packages/orbit-components/src/InputGroup/InputGroup.stories.tsx:74]: This is a story. This will never happen. The change is unnecessary.

File: packages/orbit-components/src/InputGroup/types.d.ts
Comment (DSil) [inline packages/orbit-components/src/InputGroup/types.d.ts:1]: The BC message could be improved

>BREAKING CHANGE: This prop is not used anywhere across the repository.
It is no longer implemented.

For the user, a prop usage across the repository is not relevant. What is relevant is the effect a prop might have. In this case, if the prop has no effect and is no longer meant to be used, that's what we should state.

Something like:

>BREAKING CHANGE: This prop was having no effect and is no longer meant to be used.

File: packages/orbit-components/src/Select/README.md [hunk 4]
```diff
@@ -88,3 +89,7 @@ class Component extends React.PureComponent<Props> {
   }
 }
 ```
+
+## Accessibility
+
+- The `ariaLabel` prop allows you to specify an `aria-label` attribute for the popover content component. This attribute provides additional information to screen readers, enhancing the accessibility of the component. By using `ariaLabel`, you can ensure that users who rely on assistive technologies receive the necessary context and information about the component's purpose and functionality.
```
Comment (DSil) [inline packages/orbit-components/src/Select/README.md:95]: Popover? 😅 

I feel we can be more specific here. In this case, the `aria-label` works for when we don't want a visible written `label` (because we assume the context of the select is given visually somehow) but we need to label it for screen readers.
TL;DR: The `aria-label` **must** be used when `label` is not defined, for whatever reason

File: packages/orbit-components/src/Select/types.d.ts [hunk 5]
```diff
@@ -35,4 +35,5 @@ export interface Props extends Common.Globals, Common.SpaceAfter, Common.DataAtt
   readonly insideInputGroup?: boolean;
   readonly inlineLabel?: boolean;
   readonly customValueText?: Common.Translation;
+  readonly ariaLabel?: Common.Translation;
```
Comment (DSil) [inline packages/orbit-components/src/Select/types.d.ts:38]: This should be of type string

File: packages/orbit-components/src/Select/index.tsx [hunk 6]
```diff
@@ -167,11 +169,11 @@ const Select = React.forwardRef<HTMLSelectElement, Props>((props, ref) => {
             onBlur={onBlur}
             onChange={onChange}
             tabIndex={tabIndex ? Number(tabIndex) : undefined}
-            id={selectId}
             required={required}
             ref={ref}
             aria-describedby={shown ? `${selectId}-feedback` : undefined}
             aria-invalid={error ? true : undefined}
+            aria-label={ariaLabel as string}
```
Comment (DSil) [inline packages/orbit-components/src/Select/index.tsx:176]: What happens if one does not pass `ariaLabel` prop? Does it apply `undefined` on the DOM? Or nothing at all?

**Review by DSil — CHANGES_REQUESTED** [review]

File: packages/orbit-components/src/InputGroup/InputGroup.stories.tsx (hunk 3 above)
[the remark this answers is not in the document]
Comment (sarkaaa) [reply packages/orbit-components/src/InputGroup/InputGroup.stories.tsx:74]: Without the `useEffect` implementation, value in `Select` is not changed in case I pick a different value in _Controls_ panel:

image

File: packages/orbit-components/src/Select/types.d.ts (hunk 5 above)
Comment (sarkaaa) [reply packages/orbit-components/src/Select/types.d.ts:38]: yes, I had it there before, the AI bot confused me. 🙈

File: packages/orbit-components/src/InputGroup/InputGroup.stories.tsx (hunk 3 above)
[the remark this answers is not in the document]
Comment (DSil) [reply packages/orbit-components/src/InputGroup/InputGroup.stories.tsx:74]: Hmm I see, you can leave it then

File: packages/orbit-components/src/Select/index.tsx (hunk 6 above)
Comment (sarkaaa) [reply packages/orbit-components/src/Select/index.tsx]: If `ariaLabel` is not passed, this attribute is `undefined`.

File: packages/orbit-components/src/InputGroup/index.tsx
Comment (DSil) [inline packages/orbit-components/src/InputGroup/index.tsx:1]: The commit message focuses on the implementation and not on the effect. The end user does not need to know that we use a cloneElement function internally, it's irrelevant for them. The `fix` is actually that labels on InputField and Select elements inside an InputGroup were not being propagated and now they are (as aria-label)

File: packages/orbit-components/src/InputField/index.tsx [hunk 7]
```diff
@@ -269,7 +269,7 @@ const InputField = React.forwardRef<HTMLInputElement, Props>((props, ref) => {
           ref={ref}
           tabIndex={tabIndex !== undefined ? Number(tabIndex) : undefined}
           list={list}
-          aria-labelledby={!label ? inputId : undefined}
+          aria-label={label ? (label as string) : undefined}
```
Comment (DSil) [inline packages/orbit-components/src/InputField/index.tsx:272]: This is not adding `ariaLabel` prop. The component needs to accept the prop and add it, just like you did on Select.

**Review by DSil — CHANGES_REQUESTED** [review]

File: packages/orbit-components/src/InputField/index.tsx (hunk 7 above)
Comment (sarkaaa) [reply packages/orbit-components/src/InputField/index.tsx]: fixed. 🙈
